examples/Console/TestConsoleExporter.cs

```diff
@@ -35,7 +35,7 @@ internal static object Run(ConsoleOptions options)
             // and use a custom MyProcessor, along with Console exporter.
             using var tracerProvider = Sdk.CreateTracerProviderBuilder()
                 .AddSource("MyCompany.MyProduct.MyWebServer")
-                .SetResource(Resources.CreateServiceResource("MyServiceName"))
+                .SetResource(Resource.Create("MyServiceName"))
```

Comment:
This is different from my understanding. It looks like "service" is one type of resource, and there could be other resources such as "telemetry.sdk", etc.

https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/resource/semantic_conventions#service

Reply:
How about making "service", "telemetry.sdk", etc. extension methods on Resource class? In this way each resource type can have their dedicated file/class.
e.g. `Resource.CreateServiceResource` and `Resource.CreateTelemetrySdkResource`.

Another option is `Resource.Create<Service>` but I feel the name `Service` is way to general.

Reply:
Resource.CreateServiceResource and Resource.CreateTelemetrySdkResource  - This sounds good. We can have separate classes, with each offering extension method, so discoverability is not an issue.

Reply:
I introduced a builder pattern because that felt like a good match for how we were envisioning it working and it gives extension authors a nice place to plug in there detectors.